### ISSUE TITLE
PlayedToCompletion not working properly with if_equals statement

### ISSUE THREAD

**Issue by nearcatch** [jellyfin/jellyfin-plugin-webhook#59]
Using below json, I'm trying to change the phrasing for a stop of playback. If PlayedToCompletion is 'False' I want it to say "stopped". If PlayedToCompletion is 'True' I want it to say "finished". I think the below code should work but PlayedToCompletion doesn't seem to trigger the if_equals statement. Instead I get the below blank title, but you can see in the Discord embed's description that PlayedtoCompletion's value is False and seemingly parsed properly. Not sure if I'm doing something wrong here or if this is a bug.

![image](https://user-images.githubusercontent.com/692914/133198509-b3bfbcd3-ef04-4379-83a4-c72e1fa0ae59.png)

```json
{
    "content": "{{MentionType}}",
    "avatar_url": "{{AvatarUrl}}",
    "username": "{{{BotUsername}}}",
    "embeds": [
        {
            "color": "{{EmbedColor}}",
            "footer": {
                "text": "From {{{ServerName}}}",
                "iconUrl": "{{AvatarUrl}}"
            },
            {{#if_equals ItemType 'Season'}}
                "title": "{{{NotificationUsername}}} has {{#if_equals PlayedToCompletion 'False'}}stopped{{/if_equals}}{{#if_equals PlayedToCompletion 'True'}}finished{{/if_equals}} {{{SeriesName}}} {{{Name}}}",
            {{else}}
                {{#if_equals ItemType 'Episode'}}
                    "title": "{{{NotificationUsername}}} has {{#if_equals PlayedToCompletion 'False'}}stopped{{/if_equals}}{{#if_equals PlayedToCompletion 'True'}}finished{{/if_equals}} {{{SeriesName}}} S{{SeasonNumber00}}E{{EpisodeNumber00}}: {{{Name}}}",
                {{else}}
                    "title": "{{{NotificationUsername}}} has {{#if_equals PlayedToCompletion 'False'}}stopped{{/if_equals}}{{#if_equals PlayedToCompletion 'True'}}finished{{/if_equals}} {{{Name}}} ({{Year}})",
                {{/if_equals}}
            {{/if_equals}}
            "thumbnail":{
                "url": "{{ServerUrl}}/Items/{{ItemId}}/Images/Primary"
            },
            "description": "{{{Overview}}}\nPlayed to Completion: '{{PlayedToCompletion}}'"
        }
    ]
}
```

EDIT: Don't know if it matters but my current plugin version is the one from this pull request: #57

**Comment by nearcatch** [comment]
Just to be thorough I simplified the json's title key and got the same result:
```json
{
    "content": "{{MentionType}}",
    "avatar_url": "{{AvatarUrl}}",
    "username": "{{{BotUsername}}}",
    "embeds": [
        {
            "color": "{{EmbedColor}}",
            "footer": {
                "text": "From {{{ServerName}}}",
                "iconUrl": "{{AvatarUrl}}"
            },
            "title": "{{{NotificationUsername}}} has {{#if_equals PlayedToCompletion 'False'}}stopped{{/if_equals}}{{#if_equals PlayedToCompletion 'True'}}finished{{/if_equals}} {{{Name}}} ({{Year}})",
            "thumbnail":{
                "url": "{{ServerUrl}}/Items/{{ItemId}}/Images/Primary"
            },
            "description": "{{{Overview}}}\nPlayed to Completion: '{{PlayedToCompletion}}'"
        }
    ]
}
```

**Comment by crobibero** [comment]
Please try the build attached to https://github.com/jellyfin/jellyfin-plugin-webhook/pull/61

**Comment by nearcatch** [comment]
That worked properly for the title, although now if_exist statements seem to not work. Only the Jellyfin, IMDb, and TMDb links should be displayed here. The username isn't being displayed but I assume the fork for this bug doesn't include the fix from the other thread.

![image](https://user-images.githubusercontent.com/692914/133376568-57d9463f-a365-4d59-bf63-832ea9b33e29.png)

```json
{
    "content": "{{MentionType}}",
    "avatar_url": "{{AvatarUrl}}",
    "username": "{{{BotUsername}}}",
    "embeds": [
        {
            "color": "{{EmbedColor}}",
            "footer": {
                "text": "From {{{ServerName}}}",
                "iconUrl": "{{AvatarUrl}}"
            },
            {{#if_equals ItemType 'Season'}}
                "title": "{{{NotificationUsername}}} has {{#if_equals PlayedToCompletion 'True'}}finished{{else}}stopped{{/if_equals}} {{{SeriesName}}} {{{Name}}}",
            {{else}}
                {{#if_equals ItemType 'Episode'}}
                    "title": "{{{NotificationUsername}}} has {{#if_equals PlayedToCompletion 'True'}}finished{{else}}stopped{{/if_equals}} {{{SeriesName}}} S{{SeasonNumber00}}E{{EpisodeNumber00}}: {{{Name}}}",
                {{else}}
                    "title": "{{{NotificationUsername}}} has {{#if_equals PlayedToCompletion 'True'}}finished{{else}}stopped{{/if_equals}} {{{Name}}} ({{Year}})",
                {{/if_equals}}
            {{/if_equals}}
            "thumbnail":{
                "url": "{{ServerUrl}}/Items/{{ItemId}}/Images/Primary"
            },
            "description": "{{{Overview}}}\nPlayed to Completion: '{{PlayedToCompletion}}'\n[Jellyfin]({{ServerUrl}}/web/index.html#!/details?id={{ItemId}}&serverId={{ServerId}})
            {{~#if_exist Provider_imdb~}}
            , [IMDb](https://www.imdb.com/title/{{Provider_imdb}}/)
            {{~/if_exist~}}
            {{~#if_exist Provider_tmdb~}}
                {{~#if_equals ItemType 'Movie'~}}
                    , [TMDb](https://www.themoviedb.org/movie/{{Provider_tmdb}})
                {{~else~}}
                    , [TMDb](https://www.themoviedb.org/tv/{{Provider_tmdb}})
                {{~/if_equals~}}
            {{~/if_exist~}}
            {{~#if_exist Provider_musicbrainzartist~}}
                , [MusicBrainz](https://musicbrainz.org/artist/{{Provider_musicbrainzartist}})
            {{~/if_exist~}}
            {{#if_exist Provider_audiodbartist}}
                , [AudioDb](https://theaudiodb.com/artist/{{Provider_audiodbartist}})
            {{~/if_exist~}}
            {{~#if_exist Provider_musicbrainztrack~}}
                , [MusicBrainz Track](https://musicbrainz.org/track/{{Provider_musicbrainztrack}})
            {{~/if_exist~}}
            {{~#if_exist Provider_musicbrainzalbum~}}
                , [MusicBrainz Album](https://musicbrainz.org/release/{{Provider_musicbrainzalbum}})
            {{~/if_exist~}}
            {{~#if_exist Provider_theaudiodbalbum~}}
                , [TADb Album](https://theaudiodb.com/album/{{Provider_theaudiodbalbum}})
            {{~/if_exist~}}
            {{~#if_exist Provider_tvmaze~}}
                {{~#if_equals ItemType 'Episode'~}}
                    , [TVMaze](https://www.tvmaze.com/episodes/{{Provider_tvmaze}})
                {{~/if_equals~}}
                {{~#if_equals ItemType 'Series'~}}
                    , [TVMaze](https://www.tvmaze.com/shows/{{Provider_tvmaze}})
                {{~/if_equals~}}
            {{~/if_exist~}}"
        }
    ]
}
```

**Comment by crobibero** [comment]
You're exactly correct, I forgot to pull `master` before starting the new branch. I just retested it after rebasing and it works as expected

**Comment by nearcatch** [comment]
Sorry, I mentioned in my last comment that if_exist statements stopped working with the fix for this bug. You can see above that a bunch of them are ignored.

**Comment by crobibero** [comment]
Ok, here's a new PR that I've tested and paid attention to the output...
https://github.com/jellyfin/jellyfin-plugin-webhook/pull/62

I'll hold off on releasing a new version to ensure you don't find any other oddities in the handlebars functions

**Comment by nearcatch** [comment]
Ok thanks, testing now.

As an aside, the discord template has `"iconUrl": "{{AvatarUrl}}"` for the footer. It should be `"icon_url": "{{AvatarUrl}}"`. Maybe you could fix that too instead of me opening another issue or creating a pull just for that.

**Comment by crobibero** [comment]
Well that would be why I never got the avatar url working… will PR when I get a chance

**Comment by nearcatch** [comment]
Tested the new PR and looks like everything works the way it should:
![image](https://user-images.githubusercontent.com/692914/133548426-c2834601-955e-4b23-abad-040d9991cc68.png)
